### PR TITLE
close cppl-54

### PR DESCRIPTION
**Pull request by MaksymTerekhovskyy**
- added sha1 password hash
- implemented response status check

### CODE REVIEW

File: src/Client/Controller/Controller.cpp [hunk 1]
```diff
@@ -20,14 +20,20 @@ void Controller::OnLogin(LoginInDTO in_dto)
 	LoginModel model;
 	auto request  = model.FormRequest(in_dto);
 	auto response = m_http_client.Request(request);
-	// there i should check response status
-	// but now i think that everything is ok
+
+    // checking response status
+    if(response.status >= 400)
```
Comment (AndriyFedorenko) [inline src/Client/Controller/Controller.cpp:25]: Please don't use magic number you can define constant or use Poco statuses.
https://docs.pocoproject.org/current/Poco.Net.HTTPResponse.html

File: src/Client/Models/LoginModel.cpp [hunk 2]
```diff
@@ -16,9 +16,14 @@ Net::Request LoginModel::FormRequest(LoginInDTO dto)
...
 {
 	QJsonObject json;
+
+    QByteArray encrypted_password = QByteArray::fromStdString(dto.password);
+    encrypted_password = QCryptographicHash::hash(encrypted_password, QCryptographicHash::Sha1);
+
 	json.insert("login", QString::fromStdString(dto.login));
-	json.insert("password", QString::fromStdString(dto.password));
-	return QJsonDocument(json);
+    json.insert("password", QString(encrypted_password));
```
Comment (AndriyFedorenko) [inline src/Client/Models/LoginModel.cpp:24]: Something wrong with code formatting.

File: src/Client/View/LoginPage/LoginPage.cpp [hunk 3]
```diff
@@ -14,12 +16,16 @@ LoginPage::~LoginPage()
...
+    ui->loginErrorLabel->setText(QString::fromStdString(reason));
+}
+
 void LoginPage::on_loginSubmitButton_clicked()
 {
     LoginInDTO loginInDTO;
-	loginInDTO.login	= ui->loginLineEdit->text().toStdString();
-    loginInDTO.password = ui->passwordLineEdit->text().toStdString();
-
+	loginInDTO.login	= ui->loginLineEdit->text().toStdString();
```
Comment (AndriyFedorenko) [inline src/Client/View/LoginPage/LoginPage.cpp:27]: Something wrong with code formatting. Why some many spaces between login and =

File: src/Client/View/MainWindow.h [hunk 4]
```diff
@@ -20,14 +20,12 @@ class MainWindow : public QMainWindow
...
-private:
-    Ui::MainWindow *ui;
 
-public:
     LoginPage loginPage;
     SignupPage signupPage;
     HomePage homePage;
+
+private:
+	static int PageToInt(UIPages page);
```
Comment (AndriyFedorenko) [inline src/Client/View/MainWindow.h:29]: Something wrong with code formatting.

File: src/Client/Controller/Controller.cpp (hunk 1 above)
Comment (MaksymTerekhovskyy) [reply src/Client/Controller/Controller.cpp]: fixed in the last commit

File: src/Client/View/LoginPage/LoginPage.cpp (hunk 3 above)
Comment (MaksymTerekhovskyy) [reply src/Client/View/LoginPage/LoginPage.cpp]: fixed in the last commit

File: src/Client/View/MainWindow.h (hunk 4 above)
Comment (MaksymTerekhovskyy) [reply src/Client/View/MainWindow.h]: fixed in the last commit

File: src/Client/Models/LoginModel.cpp (hunk 2 above)
Comment (MaksymTerekhovskyy) [reply src/Client/Models/LoginModel.cpp:24]: fixed in the last commit

**Review by AndriyFedorenko — APPROVED** [review]

File: src/Client/Models/LoginModel.cpp [hunk 5]
```diff
@@ -16,9 +16,14 @@ Net::Request LoginModel::FormRequest(LoginInDTO dto)
 QJsonDocument LoginModel::JSONFormatter::Format(const LoginInDTO& dto)
 {
 	QJsonObject json;
-	json.insert("login", QString::fromStdString(dto.login));
-	json.insert("password", QString::fromStdString(dto.password));
-	return QJsonDocument(json);
+
+    QByteArray encrypted_password = QByteArray::fromStdString(dto.password);
+    encrypted_password = QCryptographicHash::hash(encrypted_password, QCryptographicHash::Sha1);
```
Comment (yaroslav-kernitskyi) [inline src/Client/Models/LoginModel.cpp:21]: Storing unencrypted password in variable with name `encrypted_password` is a bit confusing.
Do you really need to store it, if you only use it once?  I suggest something like:
```
QByteArray encrypted_password = QCryptographicHash::hash(
		QByteArray::fromStdString(dto.password),
		QCryptographicHash::Sha1);
```
or
```
QByteArray unencrypted_password = QByteArray::fromStdString(dto.password);
QByteArray encrypted_password = QCryptographicHash::hash(unencrypted_password, QCryptographicHash::Sha1);
```
Also what you are doing is not really encryption, but hashing, so consider renaming variables accordingly.

File: src/Client/Controller/Controller.cpp [hunk 6]
```diff
@@ -20,14 +20,30 @@ void Controller::OnLogin(LoginInDTO in_dto)
...
+	{
+		if(response.status == Poco::Net::HTTPResponse::HTTP_UNAUTHORIZED)
+		{
+			main_window.loginPage.ChangeLoginErrorLabel(
+					"Login and password do not match");
+		}
+		else
+		{
+			QMessageBox::information(&main_window
+					, tr("Login failed!")
```
Comment (yaroslav-kernitskyi) [inline src/Client/Controller/Controller.cpp:35]: `tr` is used for translation. Will we implement it?

File: src/Client/View/LoginPage/LoginPage.h [hunk 7]
```diff
@@ -16,13 +16,16 @@ class LoginPage : public QWidget
...
+
+private:
+    Ui::LoginPage *ui;
+
+
 signals:
     void Login(LoginInDTO);
 
 private slots:
     void on_loginSubmitButton_clicked();
```
Comment (yaroslav-kernitskyi) [inline src/Client/View/LoginPage/LoginPage.h:29]: I don't like that `on_loginSubmitButton_clicked` slot is basically method, but does not follow code convention for methods. We should discuss signal/slot naming with team. Also it's better to connect button signal to this slot explicitly in constructor.

**Review by sorbeee — APPROVED** [review]

File: src/Client/Controller/Controller.cpp (hunk 6 above)
Comment (MaksymTerekhovskyy) [reply src/Client/Controller/Controller.cpp]: No, I replaced tr in the last commit

File: src/Client/View/LoginPage/LoginPage.h (hunk 7 above)
Comment (MaksymTerekhovskyy) [reply src/Client/View/LoginPage/LoginPage.h]: fixed in the last commit

I just named this slot as a regular method

File: src/Client/Models/LoginModel.cpp (hunk 5 above)
Comment (MaksymTerekhovskyy) [reply src/Client/Models/LoginModel.cpp]: Yes, we don`t need to story this password in a variable if we use it only once, but if I put QByteArray::fromStdString(dto.password) into insert method the code will look awkward and will be difficult to read so I decided not to delete this variable

In the last commit I renamed it

File: src/Client/View/LoginPage/LoginPage.h [hunk 8]
```diff
@@ -16,13 +16,16 @@ class LoginPage : public QWidget
     explicit LoginPage(QWidget *parent = nullptr);
     ~LoginPage();
 
+    void ChangeLoginErrorLabel(std::string reason);
+
+private:
+    Ui::LoginPage *ui;
```
Comment (Denys-Bykov-0) [inline src/Client/View/LoginPage/LoginPage.h:22]: Rename this variable to the m_ui, I did the same to HomePage.

**Review by Denys-Bykov-0 — CHANGES_REQUESTED** [review]

**Review by yaroslav-kernitskyi — APPROVED** [review]
I don't see any apparent issues

File: src/Client/View/LoginPage/LoginPage.h (hunk 8 above)
Comment (MaksymTerekhovskyy) [reply src/Client/View/LoginPage/LoginPage.h]: renamed in the last commit

**Review by Denys-Bykov-0 — APPROVED** [review]

**Pull request merged**